### PR TITLE
Add the "ocaml-version < 4.03" constraint on all Jane Street packages

### PR DESCRIPTION
**Pull request by diml**
Following janestreet/sexplib#17

### CODE REVIEW

File: packages/async/async.109.22.00/opam [hunk 1]
```diff
@@ -16,4 +16,7 @@ depends: [
   "async_unix" {= "109.21.00"}
   "ocamlbuild" {build}
 ]
-ocaml-version: [>= "4.00.0"]
+available: [ ocaml-version >= "4.00.0" & ocaml-version < "4.03" ]
+bug-reports: "https://github.com/janestreet/opam/issues"
```
Comment (avsm) [inline packages/async/async.109.22.00/opam]: thanks for adding all the other fields too!

**Comment by avsm** [comment]
looks good, except that `make install` is now duplicated in both `build` and `install`

File: packages/async/async.109.19.00/opam [hunk 2]
```diff
@@ -16,4 +16,7 @@ depends: [
   "async_unix" {= "109.19.00"}
   "ocamlbuild" {build}
 ]
-ocaml-version: [>= "4.00.0"]
+available: [ ocaml-version >= "4.00.0" & ocaml-version < "4.03" ]
+bug-reports: "https://github.com/janestreet/opam/issues"
+dev-repo: "https://github.com/janestreet/opam.git"
```
Comment (yallop) [inline packages/async/async.109.19.00/opam]: The `bug-reports` and `dev-repo` fields all seem to be set to `https://github.com/janestreet/opam/issues` and `https://github.com/janestreet/opam.git`.

**Comment by diml** [comment]
Thanks for pointing out, I fixed these problems

**Comment by dwightguth** [comment]
This has broken the ability to install ocaml-data-notation on 4.03.0. OPAM thinks that type_conv does not work with the latest version even though it does so it defaults back to an older version that does not work. My guess is a number of these packages probably **do** work on 4.03, so for those of us who are working on code that does not compile in 4.02 due to existing bugs, it would be nice if we could actually go through these and identify which packages work on 4.03 and which don't instead of just blanket forbidding people from using them.

**Comment by diml** [comment]
@dwightguth, sorry about that. There were too many packages to test them all, and since sexplib is not compatible with 4.03 I assumed most of our packages were not either. type_conv might be the only exception, not all version are compatible with 4.03 but the latest should be. I'm creating a PR

**Pull request merged**